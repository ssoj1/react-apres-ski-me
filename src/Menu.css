.card {
    background-color: #101010d1;
    color: #f0f2f5;
    border: 20px solid rgba(16, 16, 16, 0.63);
    box-shadow: inset 0 0 0 2px #ffffff;
    box-sizing: border-box;
    align-items: center;
}

.list-group a {
    margin-top: 1rem;
    border-left: 5px dotted;
    color: white;
}

.list-group a:not(:last-of-type) li {
    border-bottom: none !important;
}

.list-group a:hover {
    text-decoration: none;
    border-left-style: solid;
    border-left-width: 3px;
}

.list-group a:hover li {
    margin-left: 2px;
}

.list-group-item {
    color: white;
    background-color: transparent;
    text-decoration: none;
}

.list-group-item:hover {
    cursor: pointer;
}

.inner-border {
    border: 20px solid #101010d1;
    box-shadow: inset 0 0 0 10px #ffffff;
    box-sizing: border-box;
}
