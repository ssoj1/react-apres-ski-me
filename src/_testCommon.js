
// sample data to be used in testing

const snacks = [
    {
        "id": "test-snack",
        "name": "Test Snack",
        "description": "Test snack description.",
        "recipe": "Test snack recipe.",
        "serve": "Test snack serve."
      }
];

const drinks = [
    {
        "id": "test-drink",
        "name": "Test Drink",
        "description": "Test drink description.",
        "recipe": "Test drink recipe.",
        "serve": "Test drink serve."
      }
]; 

export { snacks, drinks };