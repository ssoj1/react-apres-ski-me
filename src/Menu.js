import React from "react";
import { Link } from "react-router-dom";
import "./Menu.css";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem
} from "reactstrap";

/** Menu displaying all snacks or drinks
 * 
 * Props: 
 * items - will be an array of objects of either snacks or drinks like: 
 * [{item}, ...]
 * - type - string, either "Snacks" or "Drinks"
 * 
 * Routes -> Menu 
*/

function Menu({ items, type }) {
  console.debug("* Menu", { items, type });

  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {type} Menu
          </CardTitle>
          <CardText>
            We offer a selection of {type.toLowerCase()}, made with either love
            or lack of care, depending on our mood.
          </CardText>
          <ListGroup>
            {items.map(item => (
              <Link to={`/${type.toLowerCase()}/${item.id}`} key={item.id}>
                <ListGroupItem>{item.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default Menu;
