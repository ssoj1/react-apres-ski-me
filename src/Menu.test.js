import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Menu from "./Menu";
import { snacks, drinks } from "./_testCommon";

test("renders snacks without crashing", function() {
  render(
    <MemoryRouter>
        <Menu items={snacks} type="Snacks" />
    </MemoryRouter>);
});

test("renders drinks without crashing", function() {
    render(
        <MemoryRouter>
            <Menu items={drinks} type="Drinks" />
        </MemoryRouter>);
  });

test("matches snapshot with with snacks", function() {
  const { asFragment } = render(
    <MemoryRouter>
        <Menu items={snacks} type="Snacks" />
    </MemoryRouter>);
    
  expect(asFragment()).toMatchSnapshot();
});