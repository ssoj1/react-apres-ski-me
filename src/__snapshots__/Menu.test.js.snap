// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with with snacks 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="font-weight-bold text-center card-title"
        >
          Snacks Menu
        </div>
        <p
          class="card-text"
        >
          We offer a selection of snacks, made with either love or lack of care, depending on our mood.
        </p>
        <ul
          class="list-group"
        >
          <a
            href="/snacks/test-snack"
          >
            <li
              class="list-group-item"
            >
              Test Snack
            </li>
          </a>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
